src/actions/actionFinalize.tsx

```diff
@@ -20,11 +21,16 @@ export const actionFinalize: Action = {
       }
       appState.multiElement.shape = null;
     }
+    if (!appState.elementLocked) {
+      resetCursor();
+    }
     return {
       elements: newElements,
       appState: {
         ...appState,
-        elementType: "selection",
+        elementType: appState.elementLocked
```

Comment:
```suggestion
        elementType: appState.elementLocked && appState.multiElement
```

Note, this one will require reruning prettier after applying this change